vizro-core/src/vizro/models/_dashboard.py

```diff
@@ -303,9 +304,13 @@ def _make_page_layout(self, page: Page, **kwargs):
 
     @staticmethod
     def _make_page_404_layout():
+        # The svg file is available through the _dash-component-suites/vizro route, as used in Dash's
+        # _relative_url_path, but that feels too private to access directly. Hence read the file in directly rather
+        # than referring to its path.
+        error_404_svg = base64.b64encode((VIZRO_ASSETS_PATH / "images/error_404.svg").read_bytes()).decode("utf-8")
```

Comment:
Basically never - we only do it this way because this is a special case where we have an image that's served from Vizro rather than the user assets folder. If we need to do this more in future then we should ideally have a better way of doing it because it's not very nice. But just for a one off it's ok.